### PR TITLE
FS-838-calculating-and-storing-compliance

### PR DESCRIPTION
**Pull request by gio-karunakaran**
Compliance model and methods

### CODE REVIEW

**Comment by ramsharma-prog** [comment]
Good stuff Gio 💯, some tiny comments & would it possible to add some description to the PR? 🙂

**Review by ramsharma-prog — APPROVED** [review]

File: db/models/compliance.py [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+
+class Compliance(db.Model):
+    id = db.Column(
+        "id",
+        UUIDType(binary=False),
+        default=uuid.uuid4,
+        primary_key=True,
+    )
+    created_at = db.Column("created_at", DateTime(), default=datetime.utcnow())
```
[the remark this answers is not in the document]
Comment (gio-karunakaran) [reply db/models/compliance.py:20]: yes, we want to record the time the compliance record get's created so want to call it here as it gets made 👍🏾

File: api/responses.py [hunk 2]
```diff
@@ -82,3 +83,19 @@ def scores_justifications_response_list(
         ),
         code,
     )
+
+def compliance_response(compliance: Compliance, code: int = 200):
```
[the remark this answers is not in the document]
Comment (gio-karunakaran) [reply api/responses.py:87]: I think we're just returning the make response objects from flask and response codes for every function throughout this response file, so may not need to explicitly say what this function returns?

File: api/routes/compliance/routes.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+        :param assessment_id: id of the assessment
+        :param sub_criteria_id: id of the sub_criteria
+        :return: 200 assessment JSON / 404 Error
+        """
+        try:
+            compliance_record = self.get_compliance(
+                sub_criteria_id, assessment_id
+            )
+        except ComplianceError as e:
+            return error_response(404, e.message)
```
[the remark this answers is not in the document]
Comment (gio-karunakaran) [reply api/routes/compliance/routes.py:24]: have changed this to say "error - compliance record does not exist" :)

File: api/routes/compliance/routes.py [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+                created (or an error if failure)
+        """
+        is_compliant = body.get("is_compliant")
+
+        try:
+            new_compliance = self.register_compliance(
+                sub_criteria_id, assessment_id, is_compliant
+            )
+        except ComplianceError as e:
+            return error_response(401, e.message)
```
[the remark this answers is not in the document]
Comment (gio-karunakaran) [reply api/routes/compliance/routes.py:47]: changed this to say "error - could not create compliance record" 👍🏾

**Review by adamdavies1 — APPROVED** [review]
Looks good to me

**Pull request merged**